### PR TITLE
Add peer_cert method to Ruby call object

### PR DESCRIPTION
**Pull request by ewr**
A quick PR to continue the conversation from here:

https://groups.google.com/forum/#!topic/grpc-io/58nxrxi5xUA

This adds a `peer_cert` method to the Ruby call object, which can be used by the server handler to authorize the client's access. It also allows `peer` and `peer_cert` to be used from the `ActiveCall::SingleReqView` passed as the second argument to a handler method.

`peer_cert` probably makes more sense as a more generic method for pulling property values from the call's auth context, rather than a specific method hardcoded for cert. That would work fine for our purposes, and I'm glad to change this out if that seems more acceptable.

### CODE REVIEW

**Comment by murgatroid99** [comment]
This is ok to test

**Comment by ewr** [comment]
I signed it! (via Square's corp agreement)

File: src/ruby/ext/grpc/rb_call.c [hunk 1]
```diff
@@ -214,6 +214,33 @@ static VALUE grpc_rb_call_get_peer(VALUE self) {
...
+
+  ctx = grpc_call_auth_context(call);
+
+  if (!ctx || !grpc_auth_context_peer_is_authenticated(ctx)) {
+    return Qnil;
+  }
+
+  grpc_auth_property_iterator it =
+      grpc_auth_context_find_properties_by_name(ctx, GRPC_X509_PEM_CERT_PROPERTY_NAME);
+  const grpc_auth_property *prop = grpc_auth_property_iterator_next(&it);
```
Comment (murgatroid99) [inline src/ruby/ext/grpc/rb_call.c]: Due to Ruby requirements for C code, these two lines need to be moved up with the other variable declarations (after line 221).

File: src/ruby/lib/grpc/generic/active_call.rb [hunk 2]
```diff
@@ -59,7 +59,7 @@ class ActiveCall
     include Core::CallOps
     extend Forwardable
     attr_reader(:deadline)
-    def_delegators :@call, :cancel, :metadata, :write_flag, :write_flag=
+    def_delegators :@call, :cancel, :metadata, :write_flag, :write_flag=, :peer, :peer_cert
```
Comment (murgatroid99) [inline src/ruby/lib/grpc/generic/active_call.rb]: OK, now Rubocop is unhappy about how this is aligned.

Comment (ewr) [inline src/ruby/lib/grpc/generic/active_call.rb]: I actually pushed a commit breaking that across two lines. I think that Jenkins build blew up for disk space reasons, or something like that. Should I push an empty commit to trigger a retry?

Comment (ewr) [inline src/ruby/lib/grpc/generic/active_call.rb]: Scratch that... I see where it's now complaining about the split version.

**Comment by ewr** [comment]
I just added my Square email to my github account. Not sure if that's what's keeping the CLA check from passing, or if something else needs to be done.

**Comment by murgatroid99** [comment]
LGTM

**Comment by jtattermusch** [comment]
grpc-mac-node3 failure is unrelated and other test failures are known flakes.

**Pull request merged**